e2e/single-cluster/oci_registry_test.go

```diff
@@ -0,0 +1,222 @@
+package singlecluster_test
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/rancher/fleet/e2e/testenv"
+	"github.com/rancher/fleet/e2e/testenv/kubectl"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func getFailedPodNames(k kubectl.Command, namespace string) ([]string, error) {
+	return getPodNamesByFieldSelector(k, namespace, "status.phase=Failed")
+}
+
+func getPodNamesByFieldSelector(k kubectl.Command, namespace, selector string) ([]string, error) {
+	strSelector := fmt.Sprintf("--field-selector=%s", selector)
+	out, err := k.Namespace(namespace).Get("pods", strSelector, "-o", "custom-columns=NAME:metadata.name", "--no-headers")
+	if err != nil {
+		return []string{}, err
+	}
+	return strings.Split(out, "\n"), nil
+}
+
+var _ = Describe("Single Cluster Deployments using OCI registry", Label("oci-registry", "infra-setup"), func() {
+	var (
+		asset               string
+		k                   kubectl.Command
+		tempDir             string
+		contentsID          string
+		downstreamNamespace string
+		ociRegistry         string
+	)
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		tempDir = GinkgoT().TempDir()
+		externalIP, err := k.Get("service", "zot-service", "-o", "jsonpath={.status.loadBalancer.ingress[0].ip}")
+		Expect(err).ToNot(HaveOccurred(), externalIP)
+		Expect(net.ParseIP(externalIP)).ShouldNot(BeNil())
+		ociRegistry = fmt.Sprintf("%s:5000", externalIP)
```

Comment:
Where else would you put it? BeforeSuite?
It's quite common to put setup code in BeforeEach.
Since tests can run in parallel, relying on the same name might introduce conflicts.

I'm surprised there is a Ginkgo function to get a temp dir, why don't we use it everywhere.

Reply:
Oh sure, this works anyway, no strong opinions; if we want to run tests in parallel, then my naming suggestion doesn't hold without further customisation or randomness :upside_down_face: .